notebooks/tutorials/coarse_grained_polymer.ipynb

```diff
@@ -0,0 +1,487 @@
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Creating CG Topology\n",
+    "To demonstrate the full flexibility of OpenMM we will use the Python API to define the topology. Note that you could create a PDB file and read that in instead.\n",
+    "\n",
+    "First we do the usual imports for OpenMM."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import openmm as mm\n",
+    "import openmm.app as app\n",
+    "import openmm.unit as unit\n",
+    "import numpy as np\n",
+    "from sys import stdout"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can create a new element to represent our CG bead. We set the atomic index to a large number, however the actual value is unimportant in this simulation and is just used for book-keeping purposes. We set the mass to approximate the molar mass of an amino-acid. The mass we define is important as it needs to be consistent with the chosen integration timestep."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\n",
+    "cgElement = app.Element(number=1000, name='CG-element', symbol='CG', mass=120)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We then create an empty `Topology` object."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Make an empty topology\n",
+    "topology = app.Topology()"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we can create our simulation topology using python scripting. We will define `M = 100` polymer chains which each have `N = 10` beads. As we loop over the atoms in each chain we also record the list of bonds."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Number of polymer chains\n",
+    "M = 100\n",
+    "\n",
+    "# Number of atoms in each chain\n",
+    "N = 10\n",
+    "\n",
+    "# Add each chain to the topology\n",
+    "for m in range(M):\n",
+    "\n",
+    "    # Make the chain\n",
+    "    chain = topology.addChain()\n",
+    "\n",
+    "    # Create the first residue in the chain\n",
+    "    residue = topology.addResidue(name=\"CG-residue\", chain=chain)\n",
+    "\n",
+    "    # In this example each residue is one bead so we add a single atom\n",
```

Comment:
This could be simplified a little bit.

```python
atom1 = None
for i in range(1, N):
    residue = topology.addResidue(name="CG-residue",chain=chain)
    atom2 = topology.addAtom(name="CG-bead", element=cgElement, residue=residue)

    # add the bonds in as we go
    if atom1 is not None:
        topology.addBond(atom1, atom2)
    atom1 = atom2
```

Not that it's important.  I'm just being a perfectionist. :)